src/mpl2/src/block_placement.cpp

```diff
@@ -981,32 +982,37 @@ void SimulatedAnnealingCore::CalculateNotchPenalty()
   float temp_x = 0.0;
   x_grid.push_back(x_vec[0]);
   temp_x = x_vec[0];
-  for(int i = 1; i < x_vec.size(); i++)
+  for(int i = 1; i < x_vec.size(); i++) {
     if(x_vec[i] - temp_x > 0.0) {
       temp_x = x_vec[i];
       x_grid.push_back(x_vec[i]);
     }
+  }
 
   float temp_y = 0.0;
   y_grid.push_back(y_vec[0]);
   temp_y = y_vec[0];
-  for(int i = 1; i < y_vec.size(); i++)
+  for(int i = 1; i < y_vec.size(); i++) {
     if(y_vec[i] - temp_y > 0.0) {
       temp_y = y_vec[i];
       y_grid.push_back(y_vec[i]);
     }
+  }
 
   const int num_x = x_grid.size() - 1;
   const int num_y = y_grid.size() - 1;
   vector<vector<bool> > grid(num_x);
-  for(int i = 0; i < num_x; i++)
+  for (int i = 0; i < num_x; i++) {
     grid[i].resize(num_y);
+  }
 
-  for(int i = 0; i < num_x; i++)
-    for(int j = 0; j < num_y; j++)
+  for(int i = 0; i < num_x; i++) {
+    for(int j = 0; j < num_y; j++) {
         grid[i][j] = false;
+    }
+  }
 
-  for(int i = 0; i < blocks_.size(); i++)
+  for(int i = 0; i < blocks_.size(); i++) {
```

Comment:
could be `for(const auto& block : blocks_)`

Reply:
Solved